src/OrchardCore.Modules/OrchardCore.CustomSettings/AdminMenu.cs

```diff
@@ -22,24 +24,37 @@ public class AdminMenu : INavigationProvider
 
         public async Task BuildNavigationAsync(string name, NavigationBuilder builder)
         {
-            if (!string.Equals(name, "admin", StringComparison.OrdinalIgnoreCase))
+            if (!NavigationHelper.IsAdminMenu(name))
             {
                 return;
             }
 
             foreach (var type in await _customSettingsService.GetAllSettingsTypesAsync())
             {
+                if (!_routeValues.TryGetValue(type.Name, out var routeValues))
```

Comment:
Swapping a two-property object initialization for a  lookup in a `ConcurrentDictionary` might not be a good trade (I don't know, it's just this doesn't seem like a clear win for me without measuring it). In allocations, perhaps, but not sure, and most possibly not in CPU and overall memory load.

Reply:
ConcurrentDictionary has O(1) lookup, so it should not impact performance. The idea here is so we don't create the dictionary on every request for every from every tenant. 

@sebastienros thoughts on this performance trade?

Reply:
O(1) only means constant time, not fast :). And certainly not necessarily faster than an object instantiation. It needs to be measured.

Reply:
@Piedone I expect that instantiation to be "slightly" faster since we do not need to do a lookup. On the other had I expect to have much less allocation cost.

 I did a quick benchmark to simulate a scenario where we have 10 settings entries with 1000 cycles ( simulating 1000 http requests). Here are the results 

![image](https://github.com/OrchardCMS/OrchardCore/assets/24724371/fe5fd3e8-4380-4c21-aef5-a7cec87d98b0)

But I am not getting a consistence results. When I run it again I get the following on the same scenario set.

![image](https://github.com/OrchardCMS/OrchardCore/assets/24724371/63b4b0d3-bcf6-4b8d-8565-1685e1341e97)

When I run it with 100k requests, here are the outputs I got

![image](https://github.com/OrchardCMS/OrchardCore/assets/24724371/9d41ae6c-ff9e-4b41-8dc6-46c10bc7e816)

![image](https://github.com/OrchardCMS/OrchardCore/assets/24724371/78895100-3ca2-4d13-bc73-7a83a70f651a)

The busier the app (and the longer time it runs) the more saving we get.

Reply:
@sebastienros GetOrAdd is even faster

![image](https://github.com/OrchardCMS/OrchardCore/assets/24724371/35d152f1-b20f-4b0b-a271-034182bae426)

Reply:
Yeah, benchmarks like this are what I'm talking about. Great, thanks, looks good from me then.

Reply:
Wait, in the last benchmark GetOrAdd is not faster but slower, with more allocation.

Reply:
![image](https://github.com/OrchardCMS/OrchardCore/assets/24724371/8a25bfc7-1d7c-42ca-816a-1acd6106f2be)
faster with slightly more allocation.

Reply:
That's the deviation (from the mean). You need to look at the first column.

Reply:
okay. Reverting that change